### PR TITLE
type: CheckboxGroup type infer

### PR DESCRIPTION
**Pull request by daledelv**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

1. For the four props "Change","value","defaultValue","options" of CheckboxGroup component, type inference is realized.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     CheckboxGroup type infer      |
| 🇨🇳 Chinese |      CheckboxGroup 支持范型     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

**Comment by MadCcc** [comment]
feature branch please

**Comment by li-jia-nan** [comment]
切换分支之后需要 rebase 一下，master 分支的 commit 不能出现在 feature 分支

File: components/checkbox/GroupContext.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-export interface CheckboxGroupContext {
+export interface CheckboxGroupContext<T extends CheckboxValueType = CheckboxValueType> {
   name?: string;
   toggleOption?: (option: CheckboxOptionType) => void;
   value?: any;
   disabled?: boolean;
-  registerValue: (val: string) => void;
-  cancelValue: (val: string) => void;
+  registerValue: (val: T) => void;
+  cancelValue: (val: T) => void;
```
Comment (afc163) [inline components/checkbox/GroupContext.ts:10]: Could you add a type test case to https://github.com/ant-design/ant-design/blob/master/components/checkbox/__tests__/type.test.tsx ?

Comment (li-jia-nan) [reply components/checkbox/GroupContext.ts:10]: 我加上了

**Review by afc163 — APPROVED** [review]

**Pull request merged**